File: swanious/programmers/43105/programmers-43105.py
def solution(triangle):
    n = len(triangle)
    for i in range(1, n): # 이전행의 값을 더해주기 위해 1행부터 시작
        for j in range(i + 1):
            if (j == 0): # i행의 첫번째 값
                triangle[i][j] += triangle[i-1][j]
            elif (j == i): # i행의 마지막 값
                triangle[i][j] += triangle[i-1][j-1]
            else: # i행의 중간 값
                triangle[i][j] += max(triangle[i-1][j-1], triangle[i-1][j])
                
    return max(triangle[-1])

File: HanJaehee/programmers/43105/programmers43105.java
class Solution {
    public int solution(int[][] t) {
        int len = t[t.length-1].length;
        int[][] result = new int[len][len];
        result[0][0] = t[0][0]; // 초기값 세팅
        
        for(int i=0; i<len-1; ++i){
            for(int j=0; j<t[i].length; ++j){
                result[i+1][j] = Math.max(result[i][j] + t[i+1][j], result[i+1][j]);
                result[i+1][j+1] = Math.max(result[i][j] + t[i+1][j+1], result[i+1][j+1]);
            }
        }
        
        int max = -1;
        for(int i=0; i<len; ++i){
            max = Math.max(max, result[len-1][i]);
        }
        
        // print(result, len);
        
        return max;
    }
    
    public void print(int[][] result, int len){
        for(int i=0; i<len; ++i){
            for(int j=0; j<len; ++j){
                System.out.print(result[i][j] + " ");
            }
            System.out.println();
        }
    }
}